README

```diff
@@ -41,7 +41,16 @@ shared memory mechanisms for efficient intra-node communication.
 * UCX version 1.8.0 has a bug that may cause data corruption when TCP transport
   is used in conjunction with shared memory transport. It is advised to upgrade
   to UCX version 1.9.0 and above. UCX versions released before 1.8.0 don't have
-  this bug.   
+  this bug.
+
+* UCX may hang with glibc versions 2.25-2.29 due to known bugs in pthread_rwlock
+  functions. When such hang occurs, one of UCX threads stucks in
+  pthread_rwlock_rdlock (called by ucs_rcache_get), while no other thread is
+  holding this lock. Related issue is reported in
+  [glibc Bug 23844](https://sourceware.org/bugzilla/show_bug.cgi?id=23844).
+  If this issue occurs, it is advised to use glibc version above 2.29 or
+  below 2.25.
```

Comment:
q/minor: maybe add that user can disable RCACHE (performance-sensitive)?

Reply:
i'd not recommend that, since it would kill performance in most of the cases (for instance it was not an option in two cases where we faced this issue)

Reply:
@brminich - Few suggestions. Let's clarify who is impacted by this bug. AFAIK not all transports dependent on the r-cache flow. I suspect that only infiniband/roce users are impacted ? Also, it seems that distros backported fixes for this issues. So maybe we shell highlight this - suggest to update to latest revision on glibc ? Also, it seems like the users that got impacted are those that don't use distro provided updates and build the library from vanilla source code.

Reply:
rcache is also used by sm (knem, xpmem) and gpu transports, so I'd not mention specific transports to keep the message simpler. I know about one example where glibc 2.28 was used. So are you suggesting to replace the latest sentence?

Reply:
@brminich - I actually suggest explicitly listed impacted transport (full name of the transport and well TL parameter "name"). I remember somebody mentioning that the issues pops up only for glibc versions build directly from the repo and not what is distributed. Is this correct ?

Reply:
This failure is quite seldom occasion even with impacted transports, so imo, better not mention any transport, but mention the symptom instead. AFAIU application has to have many threads (just a suggestion) and probably some other preconditions should be met for this issue to pop-up

Reply:
I agree that such races are rare but in terms of symptoms I don't believe people will be running gdb attach to figure out what is going on. So I think transport + symptoms will narrow this down sufficient enough.